synapse/storage/schema/__init__.py

```diff
@@ -133,3 +133,17 @@
 This value is stored in the database, and checked on startup. If the value in the
 database is greater than SCHEMA_VERSION, then Synapse will refuse to start.
 """
+
+BACKGROUND_UPDATES_COMPAT_VERSION = (
+    # The replace_stream_ordering_column from 6001 must have run.
+    61
+)
```

Comment:
Thinking on it more, I wonder if this should not be *just* background updates, but also refuse to start if your database is older than this? Effectively making a minimum supported version to upgrade from?